ipc/provider/src/checkpoint.rs

```diff
@@ -52,6 +63,28 @@ impl<T: BottomUpCheckpointRelayer> BottomUpCheckpointManager<T> {
             parent_handler,
             child_handler,
             finalization_blocks: 0,
+            metrics: PrometheusMetrics {
+                latest_accepted_checkpoint_height: register_int_gauge!(
+                    "latest_accepted_checkpoint_height",
+                    "Block height of latest accepted bottom-up checkpoint"
+                )
+                .unwrap(),
+                checkpoint_submit_count: register_counter!(
+                    "checkpoint_submit_count",
+                    "Number of checkpoint submission"
+                )
+                .unwrap(),
+                checkpoint_submit_succeed_count: register_counter!(
+                    "checkpoint_submit_succeed_count",
+                    "Number of successful checkpoint submission"
+                )
+                .unwrap(),
+                checkpoint_submit_fail_count: register_counter!(
+                    "checkpoint_submit_fail_count",
+                    "Number of failed checkpoint submission"
+                )
+                .unwrap(),
+            },
```

Comment:
Let's use the same design as in main:

https://github.com/consensus-shipyard/ipc/blob/147180ce386c9d950d04dcbad2a6ca19e3a1ff60/fendermint/app/src/cmd/run.rs#L312

Let's move the `metrics!` macro somewhere common, e.g. inside a new `ipc-metrics` crate under `ipc/metrics`.

Then invoke the macro from within the `checkpoint` module to define the lazy statics.

Then invoke `ipc_provider::checkpoint::register_metrics()` from the relayer command handler.

We may refactor all of this going forward, but it'll be easier if all metrics producers use the same pattern.

Reply:
Yes that's exactly I was thinking. Currently the `event` => `metric` binding is specific to fendermint app. If we want to re-use it, we need to refactor existing code to make those macro a shared lib. If that's a more preferred approach I can definitely change it into that.

Reply:
Refactored as you mentioned.